### PR TITLE
Implement config options and user credentials setting from iOS native…

### PR DESCRIPTION
**Pull request by molnartimi**
**Origin behavior**:

- User has to add her email and password in a login popup to join the videoconf.
- Jitsi reads conference config from config.js file on remote server.

**New behavior**:

- New properties in native iOS interfaces to add necessary data:
    - userId: we use 'userId@domain.com' as email to login, (domain is set as part of config)
    - password: we use generated JTW token as password
    - configJsonString: we can add config options as a JSON string (currently used options are in JitsiMeetBridgeServiceBase methods).
- Using this new information to automatically login user and set conference config.

### CODE REVIEW

File: react/features/app/components/App.native.js [hunk 1]
```diff
@@ -105,6 +111,21 @@ export class App extends AbstractApp {
                 }
             }
 
+            // SONRISA MODIFICATION: handle config set by native app
```
Comment (bruderbarna) [inline react/features/app/components/App.native.js]: I don't think it's needed to state this is a sonrisa modification, it'll be clear when doing a blame

File: react/features/app/components/App.native.js [hunk 2]
```diff
@@ -105,6 +111,21 @@ export class App extends AbstractApp {
...
+            if (typeof this.props.configJsonString !== 'undefined') {
```
Comment (bruderbarna) [inline react/features/app/components/App.native.js]: Wouldn't it be better to check if it's a string?

File: react/features/authentication/middleware.js [hunk 3]
```diff
@@ -114,7 +115,16 @@ MiddlewareRegistry.register(store => next => action => {
                 && error.name === JitsiConnectionErrors.PASSWORD_REQUIRED
                 && typeof error.recoverable === 'undefined') {
             error.recoverable = true;
-            store.dispatch(_openLoginDialog());
+
+            // SONRISA MODIFICATION: instead of show login form, login user with credentials set by native app
```
Comment (bruderbarna) [inline react/features/authentication/middleware.js]: Again, I don't think it's necessary to state that this is a modification by sonrisa

**Review by zongolista — APPROVED** [review]

File: react/features/authentication/middleware.js [hunk 4]
```diff
@@ -114,7 +115,16 @@ MiddlewareRegistry.register(store => next => action => {
...
+            const { jid, password } = _getCredentials(store.getState());
+
+            if (jid && password) {
+                logger.info('Try to login with JID', jid);
```
Comment (bruderbarna) [inline react/features/authentication/middleware.js]: Is this log message useful now that you know the code works? Seems like it's a debug log. My problem with it is that here, only an action dispatch happens, we're not actually trying to log in here. I'd be fine with keeping it if you think it's useful, and you reword it to something like: 'Dispatching connect action with JID: '

**Review by bj603 — APPROVED** [review]

**Review by kecskesk — APPROVED** [review]

**Review by bruderbarna — APPROVED** [review]
